### ISSUE TITLE
fix(html-parser): boolean attribute name_span includes trailing ">"

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3545]
## Describe the bug

In `crates/markuplint-dom/src/html_builder.rs`, the `name_span` for boolean attributes (e.g., `controls` in `<audio controls>`) includes the trailing `>` character.

This causes `MLASTHTMLAttr.name.raw` to be `"controls>"` instead of `"controls"`.

## Impact

`extract_attribute_names()` in `permitted_contents.rs` was using `html_attr.name.raw`, which included the `>`. This broke attribute selector matching (e.g., `audio[controls]` never matched because the attribute name was `"controls>"` not `"controls"`).

**Workaround applied**: Changed `extract_attribute_names()` to use `html_attr.node_name` (which is set correctly from `attr.name.clone()` in html_builder L119) instead of `html_attr.name.raw`.

## Root cause

The Rust HTML parser's attribute `name_span` end offset likely includes the `>` character when no `=` follows the attribute name (boolean attributes).

## Files

- `crates/markuplint-dom/src/html_builder.rs` L121: `name: make_token(source, attr.name_span)`
- `crates/markuplint-rules/src/rules/permitted_contents.rs` L824: workaround using `node_name`

## Reproduction

```rust
// Parse <audio controls>
// attr.name_span should cover "controls" (offset 7..15)
// but actually covers "controls>" (offset 7..16)
```

**Comment by YusukeHirao** [comment]
Workaround merged in PR #3546 (`extract_attribute_names` now uses `node_name` instead of `name.raw`). Root cause in `html_builder` `name_span` still needs fixing.

**Comment by YusukeHirao** [comment]
Fixed in PR #3550. Root cause: tokenizer `finish_attribute()` used current cursor position (past `>`) as `name_end` for boolean attributes. Fix records the name end position when transitioning out of `AttributeName` state.
